docs/topics/gradle/gradle-compiler-options.md

```diff
@@ -15,51 +15,108 @@ in the [Working with command-line compiler](command-line.md) tutorial.
 
 Kotlin compilers have a number of options for tailoring the compiling process.
 
-Using a build script, you can specify additional compilation options. Use the `compilerOptions` property of a Kotlin compilation task for it. 
-For example:
+When you use the Kotlin Gradle plugin in your project, the Gradle DSL allows comprehensive configuration of compiler options. 
+The Gradle DSL is available for [Kotlin Multiplatform](multiplatform-dsl-reference.md) and [JVM/Android](#target-the-jvm) projects.
 
-<tabs group="build-script">
-<tab title="Kotlin" group-key="kotlin">
+With the Gradle DSL, you can configure compiler options within the build script at three levels: 
+* **Extension level,** as a top-level block. These compiler options are common for all targets and shared source sets.
+* **Target level**, in the target block. These compiler options are for a specific target.
+* **Compilation unit level,** usually as a specific compilation task.
+
+![Kotlin compiler options levels](compiler-options-levels.svg){width=700}
+
+The settings at a higher level are used as a convention (default) for a lower level:
+
+* The values of extension compiler options are the default for target compiler options, including shared source sets
```

Comment:
I don't think we introduce the terms "extension compiler options" and "target compiler options" — to avoid confusion I would go the long way of "options set at extension level" and "options set at target level". This also would fix the phrasing of "...options, including source sets" which can be hard to decipher.